kylie/snippets/form.liquid

```diff
@@ -46,6 +46,7 @@
         id="email"
         type="email"
         name="form[email]"
+        pattern="/^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$/"
```

Comment:
This regex might be a bit overzealous. There's quite a few [top level domains](https://en.wikipedia.org/wiki/List_of_Internet_top-level_domains) that have more than 4 characters. I saw issues like this in other apps, where clients would be complaining that they can't register with `@something.agency` domain or something weird like that.

There are also some alphanumeric characters which are allowed in local-part of the email (I personally used + sometimes, it's quite a nice feature of gmail).

This is a very difficult topic, as email format rules are extremely complicated and often not properly supported, real-world usage is all over the place.

Reply:
> "+" is not allowed in email

[Wikipedia](https://en.wikipedia.org/wiki/Email_address) says that "local part" can contain character `+`

Reply:
That feature is part of the [RFC 5233](https://www.rfc-editor.org/rfc/rfc5233.html). But even without that feature, `+` is a valid character of the local part of email address.

Reply:
Any suggestion here about better regexp?

Reply:
Best to talk to @hrmndb about how flexible we want to be.
IMO allowing `+` in local part and not restricting top level domain length is enough.

It would still consider a lot of valid emails as invalid - but I believe a very few people uses emails like that. (Just for fun, this is a valid email address `"very.(),:;<>[]\".VERY.\"very@\\ \"very\".unusual"@strange.example.com`)

Reply:
> "very.(),:;<>[]\".VERY.\"very@\\ \"very\".unusual"@strange.example.com

Spaces are also allowed in emails?

Reply:
@dabrorius what about something like this `/^\S+@([\w-]+\.)+[\w-]{2,16}$/`?

Reply:
I think that's ok, more strict than what we had but still quite permissive 👍

Reply:
Trying to keep the balance between the lightest and strongest